packages/peregrine/lib/store/actions/cart/asyncActions.js

```diff
@@ -100,12 +93,10 @@ export const addItemToCart = (payload = {}) => {
             await dispatch(toggleDrawer('cart'));
             dispatch(actions.addItem.receive());
         } catch (error) {
-            const { response, noCartId } = error;
-
             dispatch(actions.addItem.receive(error));
 
-            // check if the cart has expired
-            if (noCartId || (response && response.status === 404)) {
+            // If we got an error and it's not a network problem, retry.
+            if (!error.networkError) {
```

Comment:
We may need to get a list of the possible graphql errors. I'm sure there may be some where we don't want to fully recreate the cart. It would suck to get some "out of stock" type error and lose your entire cart.